lib/redmine_s3/attachment_patch.rb

```diff
@@ -30,12 +31,44 @@ def put_to_s3
       end
 
       def delete_from_s3
-        logger.debug("Deleting #{disk_filename}")
-        RedmineS3::Connection.delete(disk_filename)
+        logger.debug("Deleting #{disk_filename_s3}")
+        RedmineS3::Connection.delete(disk_filename_s3)
       end
 
       # Prevent file uploading to the file system to avoid change file name
       def files_to_final_location; end
+
+      # Returns the full path the attachment thumbnail, or nil
+      # if the thumbnail cannot be generated.
+      def thumbnail_s3(options={})
```

Comment:
Can you be more specific? To be honest I based this method on attachment#thumbnail.